### ISSUE TITLE
Erratic Rotation Regardless of Path or Override

### ISSUE THREAD

**Issue by nisunku** [mjansen4857/pathplanner#624]
**Describe the bug**
We've set several autonomous routines using the PathPlanner UI, each comprised of several paths. In each path, we've set rotation targets and the robot rotation and heading of each waypoint. However, no matter what we set, while the translation of the robot behaves as expected, it rotates completely erratically, with no observable relationship to the expected rotation.

We have even tried overriding the rotation using the Holonomic Controller static method as stated in the documentation to no avail.

Lastly, we tried setting the rotation PID values to zero inside the AutoBuilder ConfigureHolonomic method in the drive subsystem constructor.  The robot continued to rotate in an unpredictable way.  This confused us the most.

**To Reproduce**
1. Use the code provided in the FRC 2024 file in the Penguin Empire repository (linked below)
- https://github.com/PenguinEmpire/FRC2024
2. Use the routine w/paths shown below
3. Robot rotation behaves unpredictably 

**Expected behavior**
The robot rotates according to the set rotation targets and angles in the path.

**Screenshots**
<img width="1278" alt="PNG image" src="https://github.com/mjansen4857/pathplanner/assets/93169793/9deeb251-06c5-4c57-9efa-6ae26f54dd41">
<img width="1280" alt="PNG image" src="https://github.com/mjansen4857/pathplanner/assets/93169793/b9645bae-9a73-4cec-8010-7456a89ee554">
<img width="1284" alt="PNG image" src="https://github.com/mjansen4857/pathplanner/assets/93169793/de68b5f1-36cc-4724-9aa1-5050c35cb3ac">
<img width="1279" alt="PNG image" src="https://github.com/mjansen4857/pathplanner/assets/93169793/85493c9e-6fcf-41c0-999b-8a1a26c81653">

**Versions: (please complete the following information):**
 - OS: MacOS 12.7
 - GUI Version: 2024.1.2
 - PPLib Version: 2024.1.2
 - PPLib Language: Java

We would greatly appreciate any help with this; we have tried everything we can think of, but nothing seems to fix it.

**Comment by bruingineer** [comment]
What does the `0.15` in the auto builder mean? I believe this argument is for the maximum radius from the center of the robot to where the wheel touches the ground (looking from the top down). 

Looking at your wheel base and track width, this should be around `sqrt(15*15+11.58*11.58) inches to meters = 0.4813`, plus or minus some inches depending on which swerve modules you’re using.

**Comment by nisunku** [comment]
Thank you for noticing this! However, we think that this won't affect the rotation. 

Update: We have switched to making our own MoveCommand for autonomous, so if you are looking at the current code from the repo, there won't be any PathPlanner code. I'm providing the link to the commit when the PathPlanner code was still there, so if anyone would like to look at it, please feel free (we still want to use this in the future for our second comp)!

[https://github.com/PenguinEmpire/FRC2024/commit/1acbb15a72f33aa06e5886e76a70f6d4edef420b](url)

**Comment by bruingineer** [comment]
This will definitely affect rotation. I know because we made the same change, but in the other direction. We set the max radius to a way to big number because of a unit conversion, and the robot barely would rotate. 

In this case, you have a smaller number set, so it would seem that it’s cause much larger rotations than needed. And then the rotation PID is trying to correct causing oscillations. 

I would recommend making that number correct and run the PathPlanner auto again.
